File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/service/dtos/CreateServiceScheduleRequest.java
package com.slotify.backend.spring.service.dtos;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.DayOfWeek;
import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class CreateServiceScheduleRequest {

    @NotNull
    private DayOfWeek dayOfWeek;

    @NotNull
    private LocalTime startTime;

    @NotNull
    private LocalTime endTime;
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/service/CreateServiceScheduleRequest.kt
package com.raj.slotify.dtos.service

import com.google.gson.annotations.SerializedName
import java.time.DayOfWeek

data class CreateServiceScheduleRequest(

    @SerializedName("dayOfWeek")
    val dayOfWeek: DayOfWeek,

    @SerializedName("startTime")
    val startTime: String,

    @SerializedName("endTime")
    val endTime: String

)
